### ISSUE TITLE
Message corruption detected.  Please disable add-ons that might modify messages on channel 5

### ISSUE THREAD

**Issue by Tetsemi** [AIE-Guild/GreenWall#202]
Getting the following message trying to link a mount in guild chat. Disabled all addons except Greenwall and still getting it.

```
"2023-01-19 19:55:20 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua:203:?] type=GUILD, message=\"\"", -- [103]
	"2023-01-19 19:55:30 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua\"]:238: in function `GreenWall_OnEvent'\n[string \"*GreenWall.xml:33_OnEvent:1:GreenWall_OnEvent] event occurred; event=GUILD_ROSTER_UPDATE", -- [104]
	"2023-01-19 19:55:30 -- debug/5 [@Interface/AddOns/GreenWall/Config.lua:148:load] configuration valid; skipping.", -- [105]
	"2023-01-19 19:55:30 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua\"]:238: in function `GreenWall_OnEvent'\n[string \"*GreenWall.xml:33_OnEvent:1:GreenWall_OnEvent] event occurred; event=GUILD_ROSTER_UPDATE", -- [106]
	"2023-01-19 19:55:30 -- debug/5 [@Interface/AddOns/GreenWall/Config.lua:148:load] configuration valid; skipping.", -- [107]
	"2023-01-19 19:55:32 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua\"]:238: in function `GreenWall_OnEvent'\n[string \"*GreenWall.xml:33_OnEvent:1:GreenWall_OnEvent] event occurred; event=GUILD_ROSTER_UPDATE", -- [108]
	"2023-01-19 19:55:32 -- debug/5 [@Interface/AddOns/GreenWall/Config.lua:148:load] configuration valid; skipping.", -- [109]
	"2023-01-19 19:55:48 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua\"]:238: in function `GreenWall_OnEvent'\n[string \"*GreenWall.xml:33_OnEvent:1:GreenWall_OnEvent] event occurred; event=GUILD_ROSTER_UPDATE", -- [110]
	"2023-01-19 19:55:48 -- debug/5 [@Interface/AddOns/GreenWall/Config.lua:148:load] configuration valid; skipping.", -- [111]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua:203:?] type=GUILD, message=\"|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTII`IQ\\\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\\\IiePjhIigPjrIiiPkRIikPk\\\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPm`IizPmlIi{Pn^ImUSuiJPMTrJ|h[Magmashell]|h|r\"", -- [112]
	"2023-01-19 19:55:52 -- debug/3 [@Interface/AddOns/GreenWall/Channel.lua:272:send] channel=5, type=1, message=\"|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTII`IQ\\\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\\\IiePjhIigPjrIiiPkRIikPk\\\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPm`IizPmlIi{Pn^ImUSuiJPMTrJ|h[Magmashell]|h|r\"", -- [113]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:323:tl_enqueue] enqueued segment: \"C#audacia##|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTII`IQ\\\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\\\IiePjhIigPjrIiiPkRIikPk\\\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPm`IizPmlIi\"", -- [114]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:340:tl_flush] servicing transmit queue; channel=5, 1 message(s) queued.", -- [115]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:126:is_connected] checking number=5, name=<<0419>>", -- [116]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:118:is_configured] number=5, name=<<0419>>, password=<<41FB>>, cversion=1, stale=false", -- [117]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:129:is_connected] confirmed number=5, name=<<0419>>", -- [118]
	"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:332:tl_dequeue] dequeued segment: \"C#audacia##|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTII`IQ\\\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\\\IiePjhIigPjrIiiPkRIikPk\\\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPm`IizPmlIi\"", -- [119]
	"2023-01-19 19:55:52 -- debug/4 [@Interface/AddOns/GreenWall/Channel.lua:354:tl_flush] channel=5, segment=\"C#audacia##|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTII`IQ\\\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\\\IiePjhIigPjrIiiPkRIikPk\\\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPm`IizPmlIi\"", -- [120]
	"2023-01-19 19:55:53 -- debug/5 [@Interface/AddOns/GreenWall/GreenWall.lua\"]:238: in function `GreenWall_OnEvent'\n[string \"*GreenWall.xml:33_OnEvent:1:GreenWall_OnEvent] event occurred; event=CHAT_MSG_CHANNEL", -- [121]
	"2023-01-19 19:55:53 -- debug/4 [@Interface/AddOns/GreenWall/Channel.lua:437:tl_receive] channel=5, sender=Tetsemi-EarthenRing, segment=\"C#audacia##\"", -- [122]
	"2023-01-19 19:55:53 -- debug/2 [@Interface/AddOns/GreenWall/Channel.lua:450:tl_receive] channel=5, tx_hash[0x893C] not found", -- [123]
	"2023-01-19 19:55:53 -- [ERROR] Message corruption detected.  Please disable add-ons that might modify messages on channel 5.", -- [124]
	"2023-01-19 19:55:53 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:459:tl_receive] opcode=C, guild_id=audacia, message=\"\"", -- [125]
	"2023-01-19 19:55:53 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:417:al_decode] type=1, message=\"\"", -- [126]
```

**Comment by mrogaski** [comment]
It looks like the message is being truncated because it's longer than the 255 character limit.  Because the item link is no longer valid, it's probably removed from the message by the game.

```
"2023-01-19 19:55:52 -- debug/3 [@Interface/AddOns/GreenWall/Channel.lua:272:send] channel=5, type=1, message=\"|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTIIIQ\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\IiePjhIigPjrIiiPkRIikPk\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPmIizPmlIi{Pn^ImUSuiJPMTrJ|h[Magmashell]|h|r\"", -- [113]
"2023-01-19 19:55:52 -- debug/5 [@Interface/AddOns/GreenWall/Channel.lua:323:tl_enqueue] enqueued segment: \"C#audacia##|cff71d5ff|Hmount:350219:100756:II\\\\IPIII]IPdII^IQLII_IQTIIIQ\\IIaIQhIYQLeoIYSLgfIYTLgjIYULgpIYVLgqIY]LhvIZPLpUIZQLpYIZRLpaIZSLpjIhgPdbIhhPd[IidPj\\IiePjhIigPjrIiiPkRIikPk\\IilPkbIimPkhIinPkoIioPkqIipPksIiqPkuIirPlJIisPlLIitPlUIiuPlZIiyPmIizPmlIi\"", -- [114]
```

**Comment by Tetsemi** [comment]
So that was linking it directly from the Mount Collection area, and I can reproduce it.

I just tested Pets, Toys, Heirlooms, and Appearances, as well as Items, and they all link fine.  It's only the Mounts that are having issues.  I tested 2 or 3 mounts from various expansions, and all have the same error.

They also show up, in chat, as a different color, not sure if that's an indication of different object type or what, just putting it here as information.
